File: final/pb13.cpp
#include <iostream>
#include <map>
using namespace std;


map<string, int> M = {
  {"dilluns",   0},
  {"dimarts",   1},
  {"dimecres",  2},
  {"dijous",    3},
  {"divendres", 4},
  {"dissabte",  5},
  {"diumenge",  6}
};


int main() {
  string a, b;
  cin >> a >> b;
  int x = M[a];
  int y = M[b];
  if (x < y - 3) x += 7;
  else if (y < x - 3) y += 7;
  cout << (x < y ? 'A' : 'B') << endl;
}

File: final/pb13.py
def DayToNumber(day):
    return ['dilluns', 'dimarts', 'dimecres', 'dijous', 'divendres', 'dissabte', 'diumenge'].index(day)

def Distance(x, y):
    return y - x if y > x else y + 7 - x

day1 = DayToNumber(input())
day2 = DayToNumber(input())

if Distance(day1, day2) > 3:
    print('B')
else:
    print('A')
